client/src/components/CustomFields.js

```diff
@@ -363,6 +366,175 @@ ReadonlyArrayObject.propTypes = {
+        <div id={`${name}-value`}>
+          <span
+            style={{ cursor: "pointer" }}
+            onClick={() => setFieldValue(name, null)}
+          >
+            <img src={REMOVE_ICON} height={14} alt="Unlink object" />
+          </span>
+          <LinkAnetEntity type={fieldValue.type} uuid={fieldValue.uuid} />
+        </div>
+      )}
+    </FastField>
+  )
+}
+AnetObjectField.propTypes = {
+  name: PropTypes.string.isRequired,
+  types: PropTypes.arrayOf(PropTypes.string),
+  formikProps: PropTypes.object
+}
+
+const ReadonlyAnetObjectField = ({ name, label, values }) => {
+  const { type, uuid } = Object.get(values, name) || {}
+  return (
+    <FastField
+      name={name}
+      label={label}
+      component={FieldHelper.ReadonlyField}
+      humanValue={
+        type &&
+        uuid && (
+          <div id={`${name}-value`}>
+            <LinkAnetEntity type={type} uuid={uuid} />
+          </div>
+        )
+      }
+    />
+  )
+}
+ReadonlyAnetObjectField.propTypes = {
+  name: PropTypes.string.isRequired,
+  label: PropTypes.string.isRequired,
+  values: PropTypes.object.isRequired
+}
+
+const ArrayOfAnetObjectsField = ({
+  name,
+  types,
+  formikProps,
+  ...otherFieldProps
+}) => {
+  const { values, setFieldValue } = formikProps
+  const fieldValue = Object.get(values, name) || []
+  return (
+    <FastField
+      name={name}
+      component={FieldHelper.SpecialField}
+      value={fieldValue}
+      widget={
+        <MultiTypeAdvancedSelectComponent
+          fieldName={name}
+          entityTypes={types}
+          isMultiSelect
+          onConfirm={(value, entityType) => {
+            if (value.length > fieldValue.length) {
+              // entity was added at the end, set correct value
+              const addedEntity = value.pop()
+              value.push({
+                type: entityType,
+                uuid: addedEntity.uuid
+              })
+            }
+            setFieldValue(name, value)
+          }}
+        />
+      }
+      {...otherFieldProps}
+    >
+      {!_isEmpty(fieldValue) && (
+        <Table id={`${name}-value`} striped condensed hover responsive>
+          <tbody>
+            {fieldValue.map(entity => (
+              <tr key={entity.uuid}>
+                <td>
+                  <span
+                    style={{ cursor: "pointer" }}
+                    onClick={() => {
+                      let found = false
+                      const newValue = fieldValue.filter(e => {
+                        if (_isEqual(e, entity)) {
+                          found = true
+                          return false
+                        }
+                        return true
+                      })
+                      if (found) {
+                        setFieldValue(name, newValue)
+                      }
+                    }}
+                  >
+                    <img src={REMOVE_ICON} height={14} alt="Unlink object" />
```

Comment:
For consistency, it would be nice if the unlink button would be displayed after the link to the object (just like it's being done for the report attendees field).

Reply:
I'll see if that looks nice enough; don't want the delete button be dangling somewhere, and we want them to line up nicely.